### PR TITLE
Shrink dom0 initramfs

### PR DESCRIPTION
**Pull request by jandryuk**
This series shrinks the compressed initramfs.gz size from 25MB to 11MB.

### CODE REVIEW

File: recipes-core/images/xenclient-initramfs-image.bb [hunk 1]
```diff
@@ -27,6 +28,15 @@ IMAGE_LINGUAS = "en-us"
...
+NO_RECOMMENDATIONS = "1"
+BAD_RECOMMENDATIONS += "busybox-syslog"
+
 PACKAGE_REMOVE = " \
     kernel-image-* \
 "
+post_rootfs_shell_commands() {
+    rm -f ${IMAGE_ROOTFS}/sbin/udhcpc;
+    rm -f ${IMAGE_ROOTFS}/sbin/ldconfig;
+    rm -rvf ${IMAGE_ROOTFS}/usr/lib/opkg;
```
Comment (eric-ch) [inline recipes-core/images/xenclient-initramfs-image.bb:40]: It looks like this comes from dom0's initramfs `POSTPROCESS`, I do not see `/usr/lib/opkg` present in the stubdomain initramfs. `opkg` seems to install `/var/lib/opkg` and `/var/cache/opkg` though.

File: recipes-core/images/xenclient-initramfs-image.bb [hunk 2]
```diff
@@ -24,10 +22,21 @@ IMAGE_INSTALL = " \
...
+inherit openxt-image
+
+NO_RECOMMENDATIONS = "1"
+BAD_RECOMMENDATIONS += "busybox-syslog"
+
+PACKAGE_REMOVE = " \
+    kernel-image-* \
+"
+post_rootfs_shell_commands() {
+    rm -f ${IMAGE_ROOTFS}/sbin/udhcpc;
```
[the remark this answers is not in the document]
Comment (jandryuk) [reply recipes-core/images/xenclient-initramfs-image.bb:38]: Looks like busybox-udhcpc only installs `/etc/udhcpc.d/50default` and `/usr/share/udhcpc/default.script` - `/sbin/udhcpc` comes in through the main busybox package.

File: recipes-core/images/xenclient-initramfs-image.bb [hunk 3]
```diff
@@ -24,10 +22,21 @@ IMAGE_INSTALL = " \
...
+    rm -f ${IMAGE_ROOTFS}/sbin/ldconfig;
```
[the remark this answers is not in the document]
Comment (jandryuk) [reply recipes-core/images/xenclient-initramfs-image.bb:39]: This works - thanks.

File: recipes-core/images/xenclient-initramfs-image.bb (hunk 2 above)
[the remark this answers is not in the document]
Comment (eric-ch) [reply recipes-core/images/xenclient-initramfs-image.bb:38]: Right, the DHCP client stuff is embedded in `busybox` and it's in our configuration.
```
root@xenclient-dom0:~# ls -l /sbin/udhcpc /usr/lib/busybox/sbin/udhcpc  
lrwxrwxrwx. 1 root root 28 Apr 13 15:20 /sbin/udhcpc -> /usr/lib/busybox/sbin/udhcpc
-rwxr-xr-x. 1 root root 21 Mar 31 07:28 /usr/lib/busybox/sbin/udhcpc
root@xenclient-dom0:~# opkg search /usr/lib/busybox/sbin/udhcpc 
busybox - 1.31.1-r0.0
root@xenclient-dom0:~# cat /usr/lib/busybox/sbin/udhcpc
#!/bin/busybox.nosuidroot@xenclient-dom0:~# 
```

File: recipes-core/images/xenclient-initramfs-image.bb (hunk 1 above)
Comment (jandryuk) [reply recipes-core/images/xenclient-initramfs-image.bb:40]: according to `du -cxsh`:
/usr/lib/opkg is 784K - lots of alternatives
/var/lib/opkg is 556K
/var/cache/opkg is 4.0K

If you remove /var/lib/opkg, you get a python backtrace when OE tries to look at /var/lib/opkg/status.  However something cleans up /var/lib/opkg and /var/cache/opkg afterwards.

/usr/lib/opkg requires manual cleanup

Comment (eric-ch) [reply recipes-core/images/xenclient-initramfs-image.bb:40]: Scratch my earlier comment, I mistakenly looked in the stubdomain initramfs instead, which has that same post-process already as mentioned...
`var/lib/opkg` and `/var/cache/opkg` would be populated by `IMAGE_FEATURE += "package-management"`, which the initramfs does not have.

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/22): [master-custom-22](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom/22/)

**Comment by jandryuk** [comment]
Pushed new version with `ldconfig` & `busybox-udhcpc` in `BAD_RECOMMENDATIONS`.  Also expanded the https://github.com/OpenXT/xenclient-oe/pull/1400/commits/c205b8e3fdac8305a1c570de8c2f55653094f3f5 commit message with more info.

I have two more minor patches on top.  Should I push them here, or PR them separately?

**Comment by eric-ch** [comment]
@jandryuk Here sounds fine, thank you.

**Comment by jandryuk** [comment]
Pushed.  This additionally uses busybox's modprobe to drop the initramfs size a little more.  Also it lets transitive dependencies pull in lvm2.

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/22/builds/24): [$master-custom-24](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom/24/)

**Review by eric-ch — APPROVED** [review]
LGTM

**Comment by eric-ch** [comment]
Will merge soon.

**Pull request merged**